### PR TITLE
Implement Optional Attendees Support with Tests

### PR DESCRIPTION
**Pull request by ptwu**
### Summary
This PR implements support for optional attendees in the Meeting Planning feature of Week 5, finishing the full functionality of the app. Optional attendees are best attempted to be included in meetings unless they block a meeting from happening entirely.

### Screenshots
N/A

### Test Plan
**Testing Rationale:** In order to best cover regular and corner cases, I tested the following 5 scenarios:
  - Optional attendee has an all-day event, making them ignored in the meeting time range selection
  - Optional attendee has a single event that rules out that time range but not the rest of time ranges.
  - There is just enough room for a meeting to occur, but optional attendee is busy during that time. They will be ignored.
  - If meeting has optional attendees only, they are treated like regular attendees.
  - Optional attendees have meetings that allow for no gaps -> no possible meetings.

Run `mvn test` in `/walkthroughs/week-5-tdd/project` to evaluate the test cases stored in `FindMeetingQueryTest.java`. See that all the tests pass (including previous ones and the 5 new ones).

### Notes
  - The diffcount of this PR is bigger mostly due to the new tests, sorry about that.
  - The frontend of the app for optional attendees does not work, and the original [repo](https://github.com/googleinterns/step/pull/38) has only recently corrected this. I may update this satellite repo, but I'm content seeing that the query method is functional from the tests.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -25,22 +25,54 @@
 public final class FindMeetingQuery {
```
Comment (xiangchengz) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: It seems all methods in this class could be static?

Comment (ptwu) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: That's true, the class state is not being used at all. Refactoring it now.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -25,22 +25,54 @@
...
 
   /**
-   * Returns a Collection of TimeRange objects representing possible intervals of a certain meeting
-   * occurring, with all participants able to go and having a certain possible duration.
-   * @param  events  A Collection of Event objects formed before the creation of this meeting
-   * @param  request A MeetingRequest object representing the meeting data.
+   * Returns a Collection of TimeRange objects representing possible intervals of
+   * a certain meeting occurring, with all participants able to go and having a
+   * certain possible duration.
```
Comment (chaowangzj) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:30]: Seems like the comment also needs to update for the optional attendees?

Comment (ptwu) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Done.
